cpp/libcudf_kafka/cmake/Modules/ConfigureGoogleTest.cmake

```diff
@@ -0,0 +1,59 @@
+set(GTEST_ROOT "${CMAKE_BINARY_DIR}/googletest")
```

Comment:
Can we not tap into the existing `cpp/cmake` which has this defined already?

Reply:
I tried that at first and really wish we could. However if you notice at [this](https://github.com/rapidsai/cudf/pull/5618/files/835027bb8a6e0f73c1c3353d815a47ba0b08b8c4#diff-1b8430a240f56383f20436c398f03d52R17) the template path depends on `${CMAKE_SOURCE_ROOT}` and therefore fails to find that file if referenced from the `libcudf_kafka` cmake file. If you have ideas about how to reuse it I would be all ears and all about making that happen but I didn't see an "easy" way to make it happen

Reply:
No I meant get rid of the entire `ConfigureGoogleTest.cmake` file and use the one in the `cpp/cmake` directory instead?

Reply:
Right, I wasn't clear but before I added this one I was doing that. However when [line](https://github.com/rapidsai/cudf/pull/5618/files/835027bb8a6e0f73c1c3353d815a47ba0b08b8c4#diff-1b8430a240f56383f20436c398f03d52R17) is ran it fails to find the template because of the `${CMAKE_SOURCE_ROOT}` I mentioned. Therefore the build fails

Reply:
Basically `"${CMAKE_SOURCE_DIR}/cmake/Templates/GoogleTest.CMakeLists.txt.cmake"` would resolve to `libcudf_kafka/cmake/Templates/GoogleTest.CMakeLists.txt.cmake` but clearly that doesn't exist unless I add it here.

Reply:
Seems like this isn't easily fixable. I will place a "TODO" comment here so we can remove in the future

Reply:
cc @millerhooks since you're touching gtest cmake module for libcudf currently